persist/fs/files.go

```diff
@@ -429,6 +725,25 @@ func NextCommitLogsFile(prefix string, start time.Time) (string, int) {
 	}
 }
 
+// NextSnapshotFileIndex returns the next snapshot file index for a given
+// namespace/shard/blockStart combination.
+func NextSnapshotFileIndex(filePathPrefix string, namespace ident.ID, shard uint32, blockStart time.Time) (int, error) {
```

Comment:
any chance to share this logic with `NextCommitLogsFile`?

Reply:
not in any obvious way, the SnapshotFiles method I'm using is pretty specific and they have different constraints due to how cleanup works